store/collection_operations.go

```diff
@@ -58,28 +58,25 @@ func (store *Store) UpdateCollectionID(ctx context.Context, path, collectionID s
 }
 
 func (store *Store) MarkCollectionPublished(ctx context.Context, collectionID string) error {
-	count, err := store.mongoCollection.Count(ctx, bson.M{fieldCollectionID: collectionID})
 	logdata := log.Data{"collection_id": collectionID}
+
+	empty, err := store.IsCollectionEmpty(ctx, collectionID)
 	if err != nil {
-		log.Error(ctx, "failed to count files collection", err, logdata)
+		log.Error(ctx, "failed to check if collection is empty", err, logdata)
 		return err
 	}
-
-	if count == 0 {
-		log.Info(ctx, "no files found in collection", logdata)
+	if empty {
+		log.Error(ctx, "collection empty check fail", ErrNoFilesInCollection, logdata)
 		return ErrNoFilesInCollection
 	}
 
-	count, err = store.mongoCollection.Count(ctx, createCollectionContainsNotUploadedFilesQuery(collectionID))
```

Comment:
think createCollectionContainsNotUploadedFilesQuery can be deleted now ?

Reply:
thanks, deleted!